# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/prim_store/mod.rs
# Commit: 548e4395ee1a
# Full Hash: 548e4395ee1a10700a729f93c0340d44ebe91ac7
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2019-09-13 15:46:31
# Regressor Bug: 1570081
# File Overlap Count: 1
# Description:
#   Bug 1570081 - Support changing the blob image visible rect in WebRender. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D39911
# ==============================================================================

diff -r 8eeddedeedb8 -r 548e4395ee1a gfx/wr/webrender/src/prim_store/mod.rs
--- a/gfx/wr/webrender/src/prim_store/mod.rs	Thu Sep 12 21:27:06 2019 +0000
+++ b/gfx/wr/webrender/src/prim_store/mod.rs	Thu Sep 12 15:54:50 2019 +0000
@@ -2381,9 +2381,12 @@
                             frame_state.gpu_cache,
                         );
                     }
-                    Some(ImageProperties { descriptor, tiling: Some(tile_size), .. }) => {
+                    Some(ImageProperties { tiling: Some(tile_size), visible_rect, .. }) => {
                         image_instance.visible_tiles.clear();
-                        let device_image_rect = DeviceIntRect::from_size(descriptor.size);
+                        // TODO: rename the blob's visible_rect into something that doesn't conflict
+                        // with the terminology we use during culling since it's not really the same
+                        // thing.
+                        let active_rect = visible_rect;
 
                         // Tighten the clip rect because decomposing the repeated image can
                         // produce primitives that are partially covering the original image
@@ -2436,7 +2439,7 @@
                             let tiles = crate::image::tiles(
                                 &layout_image_rect,
                                 &visible_rect,
-                                &device_image_rect,
+                                &active_rect,
                                 tile_size as i32,
                             );
 